source/source.go

```diff
@@ -196,6 +197,13 @@ func getProviderSpecificAnnotations(annotations map[string]string) (endpoint.Pro
 			Value: v,
 		})
 	}
+	v, exists = annotations[CloudflareCustomHostnameKey]
+	if exists {
+		providerSpecificAnnotations = append(providerSpecificAnnotations, endpoint.ProviderSpecificProperty{
+			Name:  CloudflareCustomHostnameKey,
+			Value: v,
+		})
+	}
```

Comment:
```suggestion
	if v, exists := annotations[CloudflareCustomHostnameKey]; exists {
	providerSpecificAnnotations = append(providerSpecificAnnotations, endpoint.ProviderSpecificProperty{
		Name:  CloudflareCustomHostnameKey,
		Value: v,
	})
}
```

Reply:
Looks like unit test is missing as well for this code

Reply:
Would you please elaborate on that?
All added functionality is based on the single `external-dns.alpha.kubernetes.io/cloudflare-custom-hostname` annotation, the new test function contains all possible test cases (add, replace, remove) with it:
https://github.com/kubernetes-sigs/external-dns/pull/5087/files#diff-537daf60acab77454c1d11e74a7076d095600e9b67cbf3541782c569aa162dbfR1501

Please let me know if i'm missing something, thanks!